crates/lib/src/execution_data.rs

```diff
@@ -83,7 +83,7 @@ pub enum RegisterMatrix {
 /// register.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[repr(transparent)]
-pub struct RegisterMap(HashMap<String, RegisterMatrix>);
+pub struct RegisterMap(pub HashMap<String, RegisterMatrix>);
```

Comment:
This seems fine as long as it's not possible to create invalid state with direct access to the inner map

Reply:
Yeah, technically you can't since the RegisterMatrix is built on `ndarray`, so they can't be modified to be non-rectangular, which is what would be invalid state.